modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/PageSupport.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.pagememory;
+
+import org.apache.ignite.internal.pagememory.metric.IoStatisticsHolder;
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+
+/**
+ * Supports operations on pages.
+ */
+public interface PageSupport {
+    /**
+     * Returns the page absolute pointer associated with the given page ID. Each page obtained with this method must be released by calling
+     * {@link #releasePage(int, long, long)}. This method will allocate page with given ID if it doesn't exist.
+     *
+     * @param groupId Group ID.
+     * @param pageId  Page ID.
+     * @return Page pointer.
+     * @throws IgniteInternalCheckedException If failed.
+     */
+    public long acquirePage(int groupId, long pageId) throws IgniteInternalCheckedException;
+
+    /**
+     * Returns the page absolute pointer associated with the given page ID. Each page obtained with this method must be released by calling
+     * {@link #releasePage(int, long, long)}. This method will allocate page with given ID if it doesn't exist.
+     *
+     * @param groupId    Group ID.
+     * @param pageId     Page ID.
+     * @param statHolder Statistics holder to track IO operations.
+     * @return Page pointer.
+     * @throws IgniteInternalCheckedException If failed.
+     */
+    public long acquirePage(int groupId, long pageId, IoStatisticsHolder statHolder) throws IgniteInternalCheckedException;
+
+    /**
+     * Releases page acquired by one of {@code acquirePage} methods.
+     *
+     * @param groupId Group ID.
+     * @param pageId  Page ID to release.
+     * @param page    Page pointer returned by the corresponding {@code acquirePage} call.
+     */
+    public void releasePage(int groupId, long pageId, long page);
```

Comment:
You'll be surprised, but last parameter should be called "page" to avoid confusion...
I don't think that FullPageId should be a part of low-level API, maybe we'll have utilities in the future

Reply:
> You'll be surprised, but last parameter should be called "page" to avoid confusion...

To avoid confusion with what?

Reply:
confusion with `pageAddr`